pyccel/parser/semantic.py

```diff
@@ -4149,7 +4185,10 @@ def _visit_FunctionalFor(self, expr):
             errors.report("Unrecognised output type from functional for.\n"+PYCCEL_RESTRICTION_TODO,
                           symbol=expr,
                           severity='fatal')
-
+        if target_type_name != 'list' and any(cond is not None for cond in  expr.conditions):
```

Comment:
According to its docstring `conditions` can be `None`. This line will raise an error in this case. As mentioned above I think conditions should either be non-optional or `None` should be converted to `[None]*depth` in the constructor

Reply:
You are right! I will update it!